frontend/src/views/dev_frame/load-mock-data.tsx

```diff
@@ -263,6 +275,126 @@ export function SeedDataMenu({
+
+        for (const application of seedData.applications.slice(0, limit)) {
+            const currUser = {
+                utorid: application.utorid,
+                roles: ["admin", "instructor", "ta"],
+            };
+            await dispatch(
+                debugOnlySetActiveUser(currUser, { skipInit: true })
+            );
+
+            const newApp = {
+                utorid: application.utorid,
+                student_number: application.student_number,
+                first_name: application.first_name,
+                last_name: application.last_name,
+                email: application.email,
+                phone: application.phone.toString(),
+                program: application.program,
+                department: application.department,
+                yip: application.yip,
+                gpa: application.gpa || 0,
+                previous_department_ta: application.previous_department_ta,
+                previous_university_ta: application.previous_university_ta,
+                program_start: application.program_start,
+                previous_other_university_ta:
+                    application.previous_other_university_ta,
+                position_preferences: application.position_preferences,
+            };
+
+            await apiPOST(
+                `/public/postings/${url_token}/submit`,
+                { answers: newApp },
+                true
+            );
+
+            count++;
+            setProgress(Math.round((count / total) * 100));
+        }
+
+        await dispatch(
+            debugOnlySetActiveUser({
+                utorid: initialUser.utorid,
+                roles: initialUser.roles,
+            })
+        );
+
+        setProgress(100);
+    }
+
+    async function seedInstructorPreferences(limit = 600) {
+        setStage("Instructor Preferences");
+        setProgress(0);
+
+        if (!targetSession) {
+            throw new Error("Need a valid session to continue");
+        }
+
+        let count = 0;
+        let total = seedData.applications.length;
+
+        for (const application of seedData.applications.slice(0, limit)) {
+            if (application.instructor_preferences) {
+                const targetApplication =
+                    applications.find(
+                        (currApplication) =>
+                            currApplication.applicant.utorid ===
+                            application.utorid
+                    ) || null;
+
+                if (!targetApplication) {
+                    throw new Error("No application found for " + application);
+                }
+
+                for (const position of application.instructor_preferences) {
+                    const targetPosition =
+                        positions.find(
+                            (currPosition) =>
+                                currPosition.position_code ===
+                                position.position_code
+                        ) || null;
```

Comment:
We've got cubic behaviour here...it might be worth it to create a has of position_code -> position so that this lookup can be more efficient.